src/agent/onefuzz/src/auth.rs

```diff
@@ -198,3 +237,51 @@ impl From<ManagedIdentityAccessTokenBody> for AccessToken {
         AccessToken { secret }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use anyhow::Result;
+
+    #[test]
+    fn test_managed_creds_with_valid_single_tenant() -> Result<()> {
+        let resource = "https://host-26.azurewebsites.net";
+
+        let managed_creds = ManagedIdentityCredentials::new(resource.to_string(), None)?;
+
+        assert_eq!(managed_creds.resource, resource);
+        Ok(())
+    }
+
+    #[test]
+    fn test_managed_creds_with_valid_multi_tenant_domain() -> Result<()> {
+        let resource = "https://host-26.azurewebsites.net";
+        let multi_tenant_domain = "mycloud.contoso.com";
+
+        let managed_creds = ManagedIdentityCredentials::new(
+            resource.to_string(),
+            Some(multi_tenant_domain.to_string()),
+        )?;
+
+        let expected = "https://mycloud.contoso.com/host-26";
+
+        assert_eq!(managed_creds.resource, expected);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_managed_creds_with_invalid_single_tenant() -> Result<()> {
+        let resource = "invalid-url";
+        let multi_tenant_domain = "mycloud.contoso.com";
+
+        let managed_creds = ManagedIdentityCredentials::new(
+            resource.to_string(),
+            Some(multi_tenant_domain.to_string()),
+        );
+
+        assert!(managed_creds.is_err(), true);
```

Comment:
This can be simplified:
```suggestion
        assert!(managed_creds.is_err());
```